### PR TITLE
fix(sherlock): make error message visible and guard against fetch errors

### PR DESCRIPTION
**Pull request by stdavis**

### CODE REVIEW

File: packages/utah-design-system/src/components/Sherlock.jsx [hunk 1]
```diff
@@ -59,6 +59,30 @@ let defaultItems = [
...
+    throw new Error(`error thrown during fetch to ${url}: ${error.message}`);
+  }
+
+  // handle esri response errors which return a 200 status code
+  if (responseJson.error) {
+    throw new Error(`${url} returned an error: ${responseJson.error.message}`);
+  }
+
+  return responseJson;
+}
```
Comment (steveoh) [inline packages/utah-design-system/src/components/Sherlock.jsx:84]: Do you think there is utility in making this available to all our components via @ugrc/utilities or is it specific enough for sherlock?

File: packages/utah-design-system/src/components/Sherlock.jsx [hunk 2]
```diff
@@ -361,91 +387,102 @@ export const Sherlock = (props) => {
...
-            <span
-              aria-hidden
-              className="pointer-events-none text-zinc-400 group-focus-within:text-primary-900 dark:group-focus-within:text-zinc-300"
-            >
-              <Spinner ariaLabel="searching" />
-            </span>
-          )}
-          <Button className="pr-2">
-            <ChevronsUpDownIcon
+    <Form>
```
Comment (steveoh) [inline packages/utah-design-system/src/components/Sherlock.jsx:396]: **thought**: hopefully this won't create issues if there is a need to put sherlock inside a form. I think nested forms are ok with some caveats?

File: packages/utah-design-system/src/components/Sherlock.stories.jsx [hunk 3]
```diff
@@ -87,3 +87,19 @@ export const HasMultiProvider = {
...
   ),
 };
+
+export const HasError = {
+  render: (args) => (
+    <Component
+      {...args}
+      label="Select a road"
+      placeholder="Search the roads map service"
+      provider={featureServiceProvider(
```
Comment (steveoh) [inline packages/utah-design-system/src/components/Sherlock.stories.jsx:97]: A tip for the future, the mock service worker is great for network related things like this

**Review by steveoh — APPROVED** [review]

File: packages/utah-design-system/src/components/Sherlock.jsx (hunk 1 above)
Comment (stdavis) [reply packages/utah-design-system/src/components/Sherlock.jsx:84]: I went back and forth on this and eventually decided to keep it. But if you think that it should be moved to utilities, I'm happy to move it. I also wondered if I had just duplicated some of ky and if we should switch to that.

Comment (steveoh) [reply packages/utah-design-system/src/components/Sherlock.jsx:84]: Yeah. I wobble on that also. We use fetch to have less dependencies but we're duplicating the code and use ky in almost all of our projects anyway so it's likely a free dependency.

Comment (steveoh) [reply packages/utah-design-system/src/components/Sherlock.jsx:84]: I'll resolve this for now and let's think about migrating to ky at some point across the board.

File: packages/utah-design-system/src/components/Sherlock.jsx (hunk 2 above)
Comment (stdavis) [reply packages/utah-design-system/src/components/Sherlock.jsx:396]: I'm good to wait until we hit a problem before we address this.

**Pull request merged**